### ISSUE TITLE
Dedicated search functionality in log.html

### ISSUE THREAD

**Issue by mihaiparvu** [robotframework/robotframework#3799]
Searched a bit on the issue tracker but didn't find a similar request. A colleague in my organization who wanted to start using Robot Framework asked me if there is the possibility to search the logs with regular expressions. This question immediately triggered an enhancement idea for Robot Framework, because there were a few times in which I would have found it useful to have a dedicated search functionality directly in the log.html file. The search box provided by the web browser is pretty limited in this regard as it only searches elements that are already displayed in the page. For example, if a test case is collapsed all the containing keywords are hidden and not available for search.

Having to search a huge file for a specific keyword call (that may have failed or not) can be sometimes really tedious and a dedicated search box in the bottom of the page would help a lot. I know that we already have a search button on the keyword documentation html file that helps finding keywords faster, so I think it should fit logs well.

If this is considered a valid enhancement request, I am also interested in trying to implement it.

**Comment by pekkaklarck** [comment]
I agree this would be handy. Probably wouldn't be too complicated to implement but coming up with a nice UI requires some thinking. Anyone interested to prototype this?

**Comment by stdedos** [comment]
Recap (complete quote) from the duplicate ^^^:

> I would like to be able to search for a keyword (that I know it has executed) on an execution, but I don't want to find out under which keyword it is hiding.
> 
> More ideas. Search box with:
> 
> - Search only keyword names
> - Search on Fully Qualified Keyword Names
> - Search on visible log level
> - Search on all log levels

Additionally, it would be nice if the anchor-quote did not change as part of said search; sometime it is troubling to me that an existing anchor in the URL would block from the failed keywords being expanded.

**Comment by anhtuta** [comment]
Can we merge this PR? It's very difficult for us to search for anything, we have to expand all steps in order to search! That's so annoying
